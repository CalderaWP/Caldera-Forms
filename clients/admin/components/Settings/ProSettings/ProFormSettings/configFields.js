import {optionFactory} from "../../../util/optionFactory";

export const PRO_FORM_SEND_LOCAL = 'SETTINGS/PRO/FORM/SEND_LOCAL';
export const PRO_FORM_EMAIL_LAYOUT = 'SETTINGS/PRO/FORM/EMAIL_LAYOUT';
export const PRO_FORM_PDF_LAYOUT = 'SETTINGS/PRO/FORM/PDF_LAYOUT';
export const PRO_FORM_PDF_ATATCH = 'SETTINGS/PRO/FORM/PDF_ATTACH ';
export const PRO_FORM_PDF_LINK = 'SETTINGS/PRO/FORM/PDF__LINK';

const enableOption = optionFactory(
	true,
	'Enable'
);

const disableOption = optionFactory(
	false,
	'Disable'
);

const enableDiable = [enableOption,disableOption]
export default [
	{
		id: PRO_FORM_SEND_LOCAL,
		label: 'Disable enhanced delivery for this form',
		type: 'checkbox',
		path: 'pro.send_local',
		options: [enableDiable]
	},
	{
		id: PRO_FORM_EMAIL_LAYOUT,
		label: 'Email Layout',
		type: 'dropdown',
		path: 'pro.layout',
		options: []
	},
	{
		id: PRO_FORM_PDF_LAYOUT,
		label: 'PDF Layout',
		type: 'dropdown',
		path: 'pro.pdf_layout',
	},
	{
		id: PRO_FORM_PDF_ATATCH,
		label: 'Attach PDF To Main Mailer',
		type: 'checkbox',
		path: 'pro.attach_pdf',
		options: [enableDiable]
	},
	{
		id: PRO_FORM_PDF_LINK,
		label: 'Add PDF Link',
		type: 'checkbox',
		path: 'pro.pdf_link',
		options: [enableDiable]
	},
]