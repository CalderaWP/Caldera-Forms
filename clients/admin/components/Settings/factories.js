import deepmerge from 'deepmerge';

/**
 * Provides CF Pro Local settings, with defaults merged to stored values.
 *
 * @param {Object} stored Stored values
 */
export const proLocalSettingsFactory = (stored = {} ) => {
	const defaults = {
		connected: false,
		generalSettings : {
			enhancedDelivery: false,
			generatePDFs: false,
			logLevel: 200
		},
		apiKeys: {
			public: '',
			secret: '',
			token: ''
		}

	};
	return deepmerge(
		defaults,
		stored,
	);


};