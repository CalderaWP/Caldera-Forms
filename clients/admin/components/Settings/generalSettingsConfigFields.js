import {optionFactory} from "../util/optionFactory";

export const STYLE_FORM = 'form';
export const STYLE_GRID = 'grid';
export const STYLE_ALERT = 'alert';
export const CDN_ENABLE = 'cdn';
const enableOption = optionFactory(
	true,
	'Enable'
);
export default  [
	{
		id: 'form',
		label: 'Form Styles',
		desc: 'Includes Bootstrap 3 styles on the frontend for form fields and buttons',
		type: 'fieldset',
		inputType: 'checkbox',
		path: 'generalSettings.form',
		options: [enableOption]
	},
	{
		id: 'alert',
		label: 'Alert Styles',
		desc: 'Includes Bootstrap 3 styles on the frontend for form alert notices',
		type: 'fieldset',
		inputType: 'checkbox',
		path: 'stylgeneralSettingseIncludes.alert',
		options: [enableOption]
	},
	{
		id: 'grid',
		label: 'Grid',
		desc: 'Includes Bootstrap 3 styles on the frontend for form grid layouts',
		type: 'fieldset',
		inputType: 'checkbox',
		path: 'generalSettings.grid',
		options: [enableOption]
	},{
		id: 'cdn',
		label: 'Enable Free CDN',
		desc: 'Some usage data will be shared with CDN providers',
		type: 'fieldset',
		innerType: 'checkbox',
		path: 'generalSettings.cdnEnable',
		options: [enableOption]
	},

]