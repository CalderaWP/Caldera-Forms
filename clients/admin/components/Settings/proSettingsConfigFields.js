import {optionFactory} from "../util/optionFactory";

export const PUBLIC_KEY = 'proPublicKey';
export const PRIVATE_KEY = 'proPrivateKey';
export const ENHANCED_DELIVERY = 'enhancedDelivery';
export const LOG_LEVEL = 'logLevel';
export const GENEREATE_PDFS = 'generatePDFs';
export const PRO_API_KEYS = 'apiKeys';
export const PRO_GENERAL_SETTINGS = 'generalSettings';
export const PRO_FORM_SETTINGS = 'formSettings';


export default {

	apiKeys: [
		{
			id: PUBLIC_KEY,
			label: 'Public Key',
			desc: 'Public Key For Caldera FormsSlot Pro',
			type: 'text',
			path: 'apiKeys.public',
			default: '',
		},
		{
			id: PRIVATE_KEY,
			label: 'Private Key',
			desc: 'Private Key For Caldera FormsSlot Pro',
			path: 'apiKeys.secret',
			type: 'text',
			options: [
				{
					label: 'yes',
					value: 'on'
				}
			],
			default: '',
		},
	],
	generalSettings: [
		{
			id: ENHANCED_DELIVERY,
			label: 'Enable Enhanced Delivery',
			desc: 'If enabled, messages are sent via Caldera FormsSlot Pro Servers',
			type: 'checkbox',
			path: 'generalSettings.enhancedDelivery',
			default: true,
		},
		{
			id: LOG_LEVEL,
			label: 'Minimum Log Level',
			desc: 'Setting a higher level than notice may affect performance, and should only be used when instructed by support.',
			type: 'select',
			path: 'generalSettings.logLevel',
			options: [
				optionFactory(
					100,
					'DEBUG'
				),
				optionFactory(
					200,
					'INFO'
				),
				optionFactory(
					300,
					'WARNING'
				),
				optionFactory(
					400,
					'ERROR'
				),
				optionFactory(
					500,
					'ALERT'
				),
				optionFactory(
					600,
					'EMERGENCY'
				),
			],
			default: 400,
		},
	],
	[PRO_FORM_SETTINGS]: [

	]
};





