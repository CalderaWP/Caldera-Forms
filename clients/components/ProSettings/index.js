import {ProWhatIs} from "./ProWhatIs/ProWhatIs";
import {ProEnterApp} from "./ProEnterApp/ProEnterApp";
import {ProFreeTrial} from "./ProFreeTrial/ProFreeTrial";
//import {ProFormSettings} from "./ProFormSettings/ProFormSettings";

export {
    //ProFormSettings,
    ProFreeTrial,
    ProEnterApp,
    ProWhatIs
}