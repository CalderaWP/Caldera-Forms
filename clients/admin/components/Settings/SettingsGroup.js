import React from 'react';
import PropTypes from 'prop-types';
import {object, pick} from "dot-object";
import {RenderGroup} from '@caldera-labs/components';
import {Button} from '@wordpress/components';
import classNames from 'classnames'
export class SettingsGroup extends React.PureComponent{

	constructor(props) {
		super(props);
		this.state = {
			[props.settingsKey]: props.settings
		};
		this.onSettingsChange = this.onSettingsChange.bind(this);
		this.getConfigFields = this.getConfigFields.bind(this);
		this.onSettingsSave = this.onSettingsSave.bind(this);
		this.wrapperClass = this.wrapperClass.bind(this);

	}

	/**
	 * Save the settings
	 */
	onSettingsSave() {
		this.props.onSettingsSave(this.state[this.props.settingsKey]);
	}

	/**
	 * Update internal state when settings change
	 *
	 * @param {Object} update
	 */
	onSettingsChange(update) {
		this.setState(
			{
				[this.props.settingsKey]:update
		});
	}

	/**
	 * Prepare config fields
	 *
	 * @return {Array}
	 */
	getConfigFields() {
		let currentConfigFields = this.props.configFields;
		currentConfigFields.forEach(configField => {
			const {path} = configField;
			if( undefined !== path ){
				configField.value = pick(
					path,
					this.state
				);
				configField.onValueChange = (newValue) => {
					const update = {
						...this.state,
						[path]:newValue
					};
					this.onSettingsChange(object(update));
				};
			}





		});
		return currentConfigFields;
	};

	/**
	 * Get the class for the outer element
	 * @return {String}
	 */
	wrapperClass(){
		return this.props.wrapperClass;
	}


	render(){
		if( Array.isArray( this.props.configFields ) ) {
			return(
				<div
					className={classNames(
						this.wrapperClass(),
						this.props.classNames
					)}
				>
					<RenderGroup
						configFields={this.getConfigFields()}
					/>
					<Button
						onClick={this.props.onSettingsSave}
					>
						{this.props.saveButtonText}
					</Button>
				</div>
			)
		}
		return <div>!</div>

	}

}

/**
 * Prop types for the SettingsGroup component
 * @type {{}}
 */
SettingsGroup.propTypes = {
	classNames: PropTypes.string,
	settings: PropTypes.object.isRequired,
	settingsKey: PropTypes.string.isRequired,
	onSettingsSave: PropTypes.func.isRequired,
	configFields: PropTypes.array,
	saveButtonText: PropTypes.string,
};

SettingsGroup.defaultProps= {
	saveButtonText: 'Save'
}