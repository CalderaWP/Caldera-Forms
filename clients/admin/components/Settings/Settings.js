import React from 'react';
import classNames from 'classnames'
import {TabPanel} from '@wordpress/components';
import proSettingsConfigFields from './proSettingsConfigFields';
import generalSettingsConfigFields from  './generalSettingsConfigFields';
import {SettingsGroup} from "./SettingsGroup";
import Grid from 'react-css-grid';
import {ProWhatIs} from "./ProSettings/ProWhatIs/ProWhatIs";
import {ProFreeTrial} from "./ProSettings/ProFreeTrial/ProFreeTrial";
import {ProEnterApp} from "./ProSettings/ProEnterApp/ProEnterApp";
import PropTypes from 'prop-types';

/**
 * Creates the UI for Caldera FormsSlot global settings
 */
export class Settings extends React.PureComponent {

	/**
	 * Create Settings componet
	 * @param props
	 */
	constructor(props) {
		super(props);
		this.onSettingsSave = this.onSettingsSave.bind(this);
		this.getConfigFields = this.getConfigFields.bind(this);
	}


	/**
	 * Dispatches settings to parent on save
	 * @param update
	 */
	onSettingsSave(update) {
		this.props.updateSettings(update);
	};


	getConfigFields(tabName){
		switch(tabName){
			case 'generalSettings':
				return generalSettingsConfigFields;
			case 'apiKeys':
				return proSettingsConfigFields.apiKeys;
			case 'proGeneral':
				return proSettingsConfigFields.generalSettings;
			default:
				return [];
		}
	}

	/**
	 * Creat main Caldera FormsSlot settings UI
	 * @return {*}
	 */
	render() {
		const {settings,onSettingsSave} = this.props;
		return (
			<div>
				<TabPanel
					orientation={'horizontal'}
					className={
						classNames(Settings.classNames.wrapper, this.props.classNames)
					}
					activeClass={
						classNames(Settings.classNames.active)
					}
					tabs={[
						{
							name: 'generalSettings',
							title: 'Global Form Settings',
						},
						{
							name: 'apiKeys',
							title: 'Api Keys',
							className: 'privacy-settings',
						},
						{
							name: 'proGeneral',
							title: 'Caldera Forms Pro',
							className: 'pro-general-settings',
						}
					]}
				>
					{
						(tabName) => {
							return <SettingsGroup
								settings={settings}
								settingsKey={tabName}
								onSettingsSave={onSettingsSave}
								configFields={this.getConfigFields(tabName)}
							/>
						}
					}
				</TabPanel>
				<Grid>
					{this.props.proConnected &&
						<React.Fragment>
							<ProWhatIs/>
							<ProFreeTrial/>

						</React.Fragment>
					}
					{this.props.proConnected &&
						<React.Fragment>
							<ProEnterApp/>
						</React.Fragment>
					}


				</Grid>
			</div>

		);
	}
};

Settings.propTypes = {
	generalSettings: PropTypes.shape({
		form: PropTypes.boolean,
		grid:  PropTypes.boolean,
		alert:  PropTypes.boolean,
		cdn:  PropTypes.boolean
	}),
	proSettings: PropTypes.shape({
		connected: PropTypes.boolean,
		apiKeys:  PropTypes.shape({
			proPublicKey: PropTypes.string,
			proPrivateKey: PropTypes.string,
		}),
		proGeneralSettings: PropTypes.shape({
			enhancedDelivery: PropTypes.boolean,
			logLevel: PropTypes.string
		}),
	})
};



/**
 * Class names used in the Settings component
 * @type {{wrapper: string}}
 */
Settings.classNames = {
	wrapper: 'caldera-forms-global-settings',
	active: 'caldera-forms-settings-tab-active'
};