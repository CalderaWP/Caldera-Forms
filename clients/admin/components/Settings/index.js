import {proLocalSettingsFactory} from "./factories";
import {Settings} from "./Settings";
import {SettingsGroup} from "./SettingsGroup";

/**
 *
 */
export default {
	proLocalSettingsFactory,
	Settings,
	SettingsGroup
}