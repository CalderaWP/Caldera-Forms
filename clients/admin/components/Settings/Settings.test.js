import renderer from 'react-test-renderer';
import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Settings} from "./Settings";
import {GENERAL_SETTINGS} from "./GeneralSettings/generalSettingsType";
import {STYLE_FORM} from "./GeneralSettings/configFields";
import {PRO_CONNECTED, PRO_SETTINGS} from "./ProSettings/proSettingsType";

Enzyme.configure({adapter: new Adapter()});
const handler = () => {
};

describe( 'Settings component', () => {
	it( 'Renders with minimal props', () => {
		const componet = renderer.create(
			<Settings
				onTabSelect={handler}
				onSettingsSave={handler}

			/>
		);
		expect(componet.toJSON()).toMatchSnapshot();
	});



});