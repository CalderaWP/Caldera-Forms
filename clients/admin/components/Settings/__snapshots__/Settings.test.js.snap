// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component Renders with minimal props 1`] = `
<div
  className="caldera-forms-global-settings"
>
  <div
    aria-orientation="horizontal"
    className="components-tab-panel__tabs"
    onFocus={undefined}
    onKeyDown={[Function]}
    role="tablist"
  >
    <button
      aria-controls="0-proSettings-view"
      aria-selected={true}
      className="pro-settings caldera-forms-settings-tab-active"
      id="0-proSettings"
      onClick={[Function]}
      role="tab"
      tabIndex={null}
    >
      Caldera Forms Pro
    </button>
    <button
      aria-controls="0-privacySettings-view"
      aria-selected={false}
      className="privacy-settings "
      id="0-privacySettings"
      onClick={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Privacy Settings
    </button>
    <button
      aria-controls="0-generalSettings-view"
      aria-selected={false}
      className="global-form-settings "
      id="0-generalSettings"
      onClick={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Global Form Settings
    </button>
  </div>
  <div
    aria-labelledby="0-proSettings"
    className="components-tab-panel__tab-content"
    id="0-proSettings-view"
    role="tabpanel"
    tabIndex="0"
  >
    <div
      className="caldera-forms-pro-settings"
    >
      <div
        aria-orientation="vertical"
        className="components-tab-panel__tabs"
        onFocus={undefined}
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-controls="1-apiKeys-view"
          aria-selected={true}
          className="pro-settings is-active"
          id="1-apiKeys"
          onClick={[Function]}
          role="tab"
          tabIndex={null}
        >
          Api Keys
        </button>
        <button
          aria-controls="1-whatIs-view"
          aria-selected={false}
          className="pro-what-is "
          id="1-whatIs"
          onClick={[Function]}
          role="tab"
          tabIndex={-1}
        >
          What Is Caldera FormsSlot Pro ?
        </button>
        <button
          aria-controls="1-freeTrial-view"
          aria-selected={false}
          className="pro-free-tiral "
          id="1-freeTrial"
          onClick={[Function]}
          role="tab"
          tabIndex={-1}
        >
          Free Trial
        </button>
      </div>
      <div
        aria-labelledby="1-apiKeys"
        className="components-tab-panel__tab-content"
        id="1-apiKeys-view"
        role="tabpanel"
        tabIndex="0"
      >
        <div
          className="caldera-config-field-setup"
        >
          <div
            className="caldera-config-group"
          >
            <div
              className="caldera-config-field"
            >
              <label
                htmlFor="proPublicKey"
              >
                Public Key
              </label>
              <input
                aria-describedby="proPublicKey-description"
                className="field-config field-config"
                disabled={false}
                id="proPublicKey"
                onBlur={undefined}
                onChange={[Function]}
                onFocus={undefined}
                required={false}
                type="text"
                value=""
              />
              <p
                className="description"
                id="proPublicKey-description"
              >
                Public Key For Caldera FormsSlot Pro
              </p>
            </div>
          </div>
          <div
            className="caldera-config-group"
          >
            <div
              className="caldera-config-field"
            >
              <label
                htmlFor="propPrivateKey"
              >
                Private Key
              </label>
              <input
                aria-describedby="propPrivateKey-description"
                className="field-config field-config"
                disabled={false}
                id="propPrivateKey"
                onBlur={undefined}
                onChange={[Function]}
                onFocus={undefined}
                required={false}
                type="text"
                value=""
              />
              <p
                className="description"
                id="propPrivateKey-description"
              >
                Private Key For Caldera FormsSlot Pro
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
